request.go

```diff
@@ -48,7 +48,7 @@ func HasBody(r *http.Request) bool {
 		return true
 	}
 
-	if r.Header.Get(http.CanonicalHeaderKey("content-length")) != "" {
+	if r.Header.Get("content-length") != "" {
```

Comment:
Get automatically Canonicalize the Key?

Reply:
Get itself is case-insensitive.

https://pkg.go.dev/net/http#Header.Get